core/sql/parser/sqlparser.y

```diff
@@ -6579,6 +6581,9 @@ table_reference : table_name_and_hint
               | rel_subquery_and_as_clause
                 { $$ = $1; }
 
+              | with_clause_list
```

Comment:
I agree with Hans; it would be better to move the wish_clause non-terminal to the query_expression production. Right now, that production reads:

query_expression : non_join_query_expression

You could change it to:

query_expression: with_clause non_join_query_expression
                            | non_join_query_expression,

or alternatively,

query_expression : with_clause non_join_query_expression

And then code with_clause to allow an empty production, for example:

with_clause :     { $$ = NULL; /* empty with_clause case */ }
                        |  TOK_WITH with_clause_elements